File: android/app/src/main/java/edu/gatech/cs2340/rajsjarfiles/spacetrader/entity/universe/Species.java
package edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.universe;

import java.util.Random;

/**
 * Represents species
 */
public enum Species {

    HUMANOID, // Human
    MACHINE, // Machine
    FUNGOID, // Fungi
    REPTILIAN; // Reptile

    /**
     * Get Random species that can live in certain habitats
     *
     * @param habitats habitat of the planet
     * @return habitable species
     */
    public static Species getRandomHabitableSpecies(Habitats habitats) {
        Random rand = new Random();
        return habitats.getHabitableSpecies().
                get(rand.nextInt(habitats.getHabitableSpecies().size()));
    }
}

File: ios/SpaceTrader/SpaceTrader/src/entity/universe/Species.swift
//
//  .swift
//  SpaceTrader
//
//  Created by Thomas Suarez on 4/20/19.
//  Copyright © 2019 RajsJarFiles. All rights reserved.
//

import Foundation

struct Species {
    
    public static let HUMANOID = Species()
    public static let MACHINE = Species()
    public static let FUNGOID = Species()
    public static let REPTILIAN = Species()
    
    public static var values = [
        HUMANOID,
        MACHINE,
        FUNGOID,
        REPTILIAN
    ]
    
    init() {
        
    }
    
    static func getRandomHabitableSpecies(habitats : Habitats) -> Species {
        let r = Int.random(in: 0..<values.count)
        return values[r]
    }
}
